alpha/apps/kaltura/lib/storage/urlTokenizers/kWowzaSecureTokenUrlTokenizer.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+class kWowzaSecureTokenUrlTokenizer extends kUrlTokenizer
+{
+	/**
+	 * @var string
+	 */
+	protected $paramPrefix;
+	
+	/**
+	 * @var string
+	 */
+	protected $hashAlgorithm;
+	
+	public function tokenizeSingleUrl($url, $urlPrefix = null)
+	{
+		$tokenPrefix = $this->getParamPrefix();
+		
+		$expiryTimeFrame = $this->getWindow();
+		
+		$params = array();
+		
+		$params['starttime'] = time();
+		$params['endtime'] = time() + $expiryTimeFrame;
+		
+		$urlTokenParameters = $this->prefixParams($tokenPrefix,$params);
+		
+		$urlTokenParameters[$tokenPrefix.'hash'] = $this->getHash($urlTokenParameters,$url);
+		
+		return $url."?".http_build_query($urlTokenParameters, '', '&');
+	}
+	
+	/**
+	 * Get prefixed parameters
+	 * 
+	 * @return array
+	 */
+	private function prefixParams($prefix, $rawParameters)
+	{
+		$prefixedParameters = array();
+		
+		if(empty($rawParameters))
+		{
+			return $prefixedParameters;
+		}
+		
+		foreach($rawParameters as $key => $param)
+		{
+			$prefixedParameters[$prefix.$key] = $param;
+		}
+
+		return $prefixedParameters;
+	}
+	
+	/**
+	 * Get hash token
+	 * 
+	 * @return string
+	 */
+	public function getHash($params, $url)
+	{
+		$tokenKey = $this->getKey();
+		
+		$params[$tokenKey] = "";
+		
+		if($this->getLimitIpAddress())
+		{
+			$params[self::getRemoteAddress()] = "";
+		}
+		
+		ksort($params);
+		
+		$query = '';
+		foreach($params as $k => $v)
+		{
+			$query .= '&' . $k;
+			if(isset($v) && !empty($v))
+			{
+				$query .= '=' . $v;
+			}
+		}
+		
+		$query = trim($query,'&');
+		
+		$path = '';
+		
+		if($parsed_url = parse_url($url))
```

Comment:
Since you care only about the path, better pass PHP_URL_PATH to parse_url. Also, the rest of the code here uses lower camel convention while here you use underscore (parsed_url)